d2bs/kolbot/libs/common/Misc.js

```diff
@@ -374,51 +374,28 @@ MainLoop:
 	},
 
 	// Put a skill on desired slot
-	setSkill: function (skillId, hand) {
+	setSkill: function (skillId, hand, item) {
 		// Check if the skill is already set
 		if (me.getSkill(hand === 0 ? 2 : 3) === skillId) {
 			return true;
 		}
 
-		if (!me.getSkill(skillId, 1)) {
+		if (!item && !me.getSkill(skillId, 1)) {
 			return false;
 		}
 
-		if (hand === undefined || hand === 3) {
+		// Charged skills must be cast from right hand
+		if (hand === undefined || hand === 3 || item) {
```

Comment:
Can you display a warning here if the hand specified was not zero with casting charges? It overrides the hand here, so it would be nice to let script writers know that it's doing this.

Reply:
Good idea. i'll implement that